pymc/smc/smc.py

```diff
@@ -379,31 +379,31 @@ def tune(self):
     def mutate(self):
         """Independent Metropolis-Hastings perturbation."""
         ac_ = np.empty((self.n_steps, self.draws))
-
-        cov = self.proposal_dist.cov
         log_R = np.log(self.rng.random((self.n_steps, self.draws)))
+
+        # The proposal is independent from the current point.
+        # We have to take that into account to compute the Metropolis-Hastings acceptance
+        # We first compute the logp of proposing a transition to the current points.
+        backward_logp = self.proposal_dist.logpdf(self.tempered_posterior)
```

Comment:
This should be inside the loop, self.tempered_posterior is updated at each n_step

Reply:
I update this variable inplace at the end of the loop

Reply:
Ohh I see now!